### PR TITLE
Agreement acceptance checkboxes

### PR DESCRIPTION
**Pull request by mikrise2**

### CODE REVIEW

File: ijPlugin/src/main/resources/org/jetbrains/research/tasktracker/ui/main/panel/template/agreement.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<div class="container">
```
Comment (nbirillo) [inline ijPlugin/src/main/resources/org/jetbrains/research/tasktracker/ui/main/panel/template/agreement.html:1]: Screenshot 2023-10-30 at 09 15 29
Could you please make it prettier?
1) add a margin between the form description and itself (1)
2) Add a header before a list of agreements, like "I agree to the terms and conditions:"
3) put "Optional" or "Required" right after checkbox to indicate which checkboxes are required and optional
4) make the margin between agreements smaller (2) 
5) put the agreements to the left part of the screen, not in the center
6) put a mock link in at least one of the examples

File: ijPlugin/src/main/resources/org/jetbrains/research/tasktracker/ui/main/panel/template/agreement.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    <label for="name">Please write your name: </label>
```
Comment (nbirillo) [inline ijPlugin/src/main/resources/org/jetbrains/research/tasktracker/ui/main/panel/template/agreement.html:2]: Write -> put

File: ijPlugin/src/main/resources/org/jetbrains/research/tasktracker/ui/main/panel/template/agreement.html (hunk 1 above)
Comment (nbirillo) [reply ijPlugin/src/main/resources/org/jetbrains/research/tasktracker/ui/main/panel/template/agreement.html:1]: Also, add an error popup to inform the user that the user has to agree to all required terms and conditions + personal info to start (in this state)
Screenshot 2023-10-30 at 09 21 26

File: ijPlugin/src/main/kotlin/org/jetbrains/research/tasktracker/ui/main/panel/panelStates/DefaultStates.kt [hunk 3]
```diff
@@ -15,14 +15,35 @@ import org.jetbrains.research.tasktracker.util.survey.SurveyParser
 
 typealias Panel = MainPluginPanelFactory
 
+/**
+ * A page for collecting user data, and checkboxes for user agreement acceptance.
+ */
+fun Panel.agreementAcceptance() {
+    loadBasePage(AgreementTemplate.loadCurrentTemplate(), "ui.button.next", false)
+    setNextAction {
+        checkInputs()
```
Comment (nbirillo) [inline ijPlugin/src/main/kotlin/org/jetbrains/research/tasktracker/ui/main/panel/panelStates/DefaultStates.kt:24]: You need not only to check all inputs you need to collect this data

**Review by nbirillo — CHANGES_REQUESTED** [review]

File: ijPlugin/src/main/kotlin/org/jetbrains/research/tasktracker/ui/main/panel/panelStates/DefaultStates.kt (hunk 3 above)
Comment (mikrise2) [reply ijPlugin/src/main/kotlin/org/jetbrains/research/tasktracker/ui/main/panel/panelStates/DefaultStates.kt:26]: I'm sorry, but I didn't have time to do this, I suggest leaving this for the next PR.

Comment (nbirillo) [reply ijPlugin/src/main/kotlin/org/jetbrains/research/tasktracker/ui/main/panel/panelStates/DefaultStates.kt:26]: yep, it is a great idea, let's make another PR

File: ijPlugin/src/main/kotlin/org/jetbrains/research/tasktracker/util/notifier/NotifyError.kt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package org.jetbrains.research.tasktracker.util.notifier
+
+import com.intellij.notification.NotificationGroupManager
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.project.Project
+import org.jetbrains.annotations.Nls
+
+fun notifyError(project: Project, content: @Nls String?) {
+    println("hello")
```
Comment (nbirillo) [inline ijPlugin/src/main/kotlin/org/jetbrains/research/tasktracker/util/notifier/NotifyError.kt:9]: Delete?

File: ijPlugin/src/main/kotlin/org/jetbrains/research/tasktracker/util/notifier/NotifyError.kt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    NotificationGroupManager.getInstance().getNotificationGroup("Tasktracker")
```
Comment (nbirillo) [inline ijPlugin/src/main/kotlin/org/jetbrains/research/tasktracker/util/notifier/NotifyError.kt:10]: Move Tasktracker into constants? Maybe is it better to use the plugin id here?

**Review by nbirillo — CHANGES_REQUESTED** [review]
I also sent some comments about formatting on the panel in dm

**Review by eartser — CHANGES_REQUESTED** [review]
Looks good to me! I guess I discovered a bug with links, could you please look at it once again?

File: ijPlugin/src/main/resources/org/jetbrains/research/tasktracker/config/agreement_default.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+agreements:
+  - text: |
+      Multiline
+      example
+    required: false
+  - text: "Example <a href=\"https://github.com/JetBrains-Research/tasktracker-3\">link</a>"
```
[the remark this answers is not in the document]
Comment (eartser) [reply ijPlugin/src/main/resources/org/jetbrains/research/tasktracker/config/agreement_default.yaml:6]: Alright then, everything else is fine!

**Review by eartser — APPROVED** [review]

**Pull request merged**